src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs

```diff
@@ -140,6 +140,20 @@ protected override Task CreateVersionsAsync(CancellationToken cancellationToken)
             var latestPrerelease = allVersionsAllowed.FirstOrDefault(v => v.version.IsPrerelease);
             var latestStableVersion = allVersionsAllowed.FirstOrDefault(v => !v.version.IsPrerelease);
 
+            foreach (IProjectContextInfo project in _nugetProjects)
+            {
+                if (project.ProjectStyle.Equals(ProjectModel.ProjectStyle.PackageReference) && installedDependency != null)
+                {
+                    VersionRange installedVersionRange = VersionRange.Parse(installedDependency?.VersionRange?.OriginalString, true);
+                    NuGetVersion bestVersion = installedVersionRange.FindBestMatch(allVersionsAllowed.Select(v => v.version));
+                    _versions.Add(new DisplayVersion(installedVersionRange, bestVersion, additionalInfo: null));
```

Comment:
Given that the `bestversion` is computed using `installedVersionRange`, I think we can modify the existing constructor to call to `FindBestMatch` on the `VersionRange` to figure out which version to display in the PM UI. Please correct if my understanding is incorrect here because you may have debugged this code more than I did.